onnx/shape_inference/implementation.cc

```diff
@@ -161,5 +179,98 @@ void InferShapes(
   }
 }
 
+void InferShapeForFunctionNode(
+    const FunctionProto& func,
+    const ISchemaRegistry* schema_registry,
+    InferenceContext& ctx) {
+  int domain_version = (int)func.since_version();
+  GraphProto g;
+  // Get a temproary tensor-shape map
+  std::unordered_map<std::string, TypeProto*> temp_valueTypesByName;
```

Comment:
Can we add another function called `InferShape(GraphProto g)`, so you just need to expand the node and do the shape inference?

Reply:
Our API is only for model right now, should be easy to add one for graph: https://github.com/onnx/onnx/blob/master/onnx/shape_inference/implementation.h#L91

Reply:
Sure will do so